components/Post/PostContent.js

```diff
@@ -68,9 +126,23 @@ const PostContent = (props) => {
         }}
       >
         <Container maxWidth="md">
-          <Typography variant="h3" sx={{ mt: 3, mb: 3 }}>
-            {post.title}
-          </Typography>
+          <Stack
+            direction="row"
+            justifyContent="space-between"
+            alignItems="baseline"
+          >
+            <Typography variant="h3" sx={{ mt: 3, mb: 3 }}>
+              {post.title}
+            </Typography>
+            {isLogin
```

Comment:
for this part, I think you can wrap it as a React.memo function, and use if else to make it more readable